### ISSUE TITLE
Power9 RDMACM issues

### ISSUE THREAD

**Issue by aror92** [openucx/ucx#4563]
Our group has been trying to use UCX’s client-server communication with rdmacm on Power9 with an infiniband adapter. We’ve tried it on two different Power9 clusters, one with RHEL 7.6 (Maipo), and one with Centos7. In each case, the memory domain for the Infiniband NIC seems to be incorrectly detected, and for Memory domain: rdmacm we get a `< no supported devices found >` message. Then running UCX’s server examples gives us a note that `server=(null)`, and it perpetually says `Waiting for connection…`

We’ve tried UCX versions 1.5 and 1.6.1 and have the same issue with both of them. The examples we’ve tried and are having issues with are: uct_hello_world, ucp_hello_world, ucp_client_server. The architecture we’re using is ppc64le, and we’re using rdma-core.

Does anyone have some insight on what we need to do to get rdmacm working? Output and environment variable information are below.

Our RDMACM environment variables are:
```
UCX_RDMACM_ALLOC=huge,thp,md,mmap,heap
UCX_RDMACM_FAILURE=ERROR
UCX_RDMACM_MAX_NUM_EPS=inf
UCX_RDMACM_BACKLOG=1024
UCX_RDMACM_CM_ID_QUOTA=64
```

Output from `ucx_info -d` relating to rdmacm:
```
# Memory domain: rdmacm
#     Component: rdmacm
#           supports client-server connection establishment via sockaddr
#   < no supported devices found >

```
Output from `ucx_info -d` relating to Infiniband NIC:
```
# Memory domain: mlx5_0
#     Component: ib
#             register: unlimited, cost: 90 nsec
#           remote key: 8 bytes
#           local memory handle is required for zcopy
#
#   Transport: rc_verbs
#      Device: mlx5_0:1
#
#      capabilities:
#            bandwidth: 0.00 + 11794.23 MB/sec
#              latency: 600 nsec + 1 * N
#             overhead: 75 nsec
#            put_short: <= 124
#            put_bcopy: <= 8K
#            put_zcopy: <= 1G, up to 3 iov
#  put_opt_zcopy_align: <= 512
#        put_align_mtu: <= 4K
#            get_bcopy: <= 8K
#            get_zcopy: 65..1G, up to 3 iov
#  get_opt_zcopy_align: <= 512
#        get_align_mtu: <= 4K
#             am_short: <= 123
#             am_bcopy: <= 8191
#             am_zcopy: <= 8191, up to 2 iov
#   am_opt_zcopy_align: <= 512
#         am_align_mtu: <= 4K
#            am header: <= 127
#               domain: device
#           atomic_add: 64 bit
#          atomic_fadd: 64 bit
#         atomic_cswap: 64 bit
#           connection: to ep
#             priority: 40
#       device address: 3 bytes
#           ep address: 3 bytes
#       error handling: peer failure
#
#
#   Transport: rc_mlx5
#      Device: mlx5_0:1
#
#      capabilities:
#            bandwidth: 0.00 + 11794.23 MB/sec
#              latency: 600 nsec + 1 * N
#             overhead: 40 nsec
#            put_short: <= 2K
#            put_bcopy: <= 8K
#            put_zcopy: <= 1G, up to 8 iov
#  put_opt_zcopy_align: <= 512
#        put_align_mtu: <= 4K
#            get_bcopy: <= 8K
#            get_zcopy: 65..1G, up to 8 iov
#  get_opt_zcopy_align: <= 512
#        get_align_mtu: <= 4K
#             am_short: <= 2046
#             am_bcopy: <= 8190
#             am_zcopy: <= 8190, up to 3 iov
#   am_opt_zcopy_align: <= 512
#         am_align_mtu: <= 4K
#            am header: <= 186
#               domain: device
#           atomic_add: 64 bit
#          atomic_fadd: 64 bit
#         atomic_cswap: 64 bit
#           connection: to ep
#             priority: 40
#       device address: 3 bytes
#           ep address: 7 bytes
#       error handling: buffer (zcopy), remote access, peer failure
```

Output from running ucp_hello_world:
```
$ ./ucp_hello_world
INFO: UCP_HELLO_WORLD mode = 0 server = (null) port = 13337
UCX_NET_DEVICES=all
UCX_SHM_DEVICES=all
UCX_ACC_DEVICES=all
UCX_SELF_DEVICES=all
UCX_TLS=rc,cuda_copy,cuda_ipc
UCX_ALLOC_PRIO=md:sysv,md:posix,huge,thp,md:*,mmap,heap
UCX_SOCKADDR_TLS_PRIORITY=rdmacm,*
UCX_SOCKADDR_AUX_TLS=ud
UCX_WARN_INVALID_CONFIG=y
UCX_BCOPY_THRESH=0
UCX_RNDV_THRESH=auto
UCX_RNDV_SEND_NBR_THRESH=256K
UCX_RNDV_THRESH_FALLBACK=inf
UCX_RNDV_PERF_DIFF=1.000
UCX_MAX_EAGER_RAILS=1
UCX_MAX_RNDV_RAILS=2
UCX_RNDV_SCHEME=put_zcopy
UCX_ZCOPY_THRESH=auto
UCX_BCOPY_BW=5800M
UCX_ATOMIC_MODE=guess
UCX_MAX_WORKER_NAME=32
UCX_USE_MT_MUTEX=n
UCX_ADAPTIVE_PROGRESS=y
UCX_SEG_SIZE=8K
UCX_TM_THRESH=1K
UCX_TM_MAX_BB_SIZE=1K
UCX_TM_FORCE_THRESH=8K
UCX_TM_SW_RNDV=n
UCX_NUM_EPS=auto
UCX_NUM_PPN=auto
UCX_RNDV_FRAG_SIZE=512K
UCX_MEMTYPE_CACHE=n
UCX_FLUSH_WORKER_EPS=y
UCX_UNIFIED_MODE=n
UCX_SOCKADDR_CM_ENABLE=n
[0x48860] local address length: 246
Waiting for connection...
```

**Comment by yosefe** [comment]
@aror92 did you try to run a client which would connect to the running server?

**Comment by aror92** [comment]
@yosefe Yes, we were trying that, but after seeing your reply, we took a look at the client again and realized that we weren't giving it the right IP address for the server. One of the difficulties was that the two different Infiniband switches used a common IP address which was showing up in a different place than the instructions for finding the IP address suggested. Once we noticed that and gave that IP address to the client, the examples started working. Thanks!
